### PR TITLE
Build warnings

### PR DESCRIPTION
**Pull request by DenverM80**
Eliminate most of the ~100 build warnings about defined but unused methods.

Fix Ds3Element equals comparison to handle properties that are null (ComponentType)

Special case TargetReadPreference to TargetReadPreferenceType to avoid conflict with Ds3TargetReadPreference response type.

Improve filtering of array types

     * Example:
     *      <Ds3Targets>
                <Ds3Target>
                    <AccessControlReplication>NONE</AccessControlReplication>
                    <AdminAuthId>aid</AdminAuthId>
                    <AdminSecretKey>ask</AdminSecretKey>
                    <DataPathEndPoint>dp</DataPathEndPoint>
                    <DataPathHttps>true</DataPathHttps>
                    <DataPathPort />
                    <DataPathProxy />
                    <DataPathVerifyCertificate>true</DataPathVerifyCertificate>
                    <DefaultReadPreference>LAST_RESORT</DefaultReadPreference>
                    <Id>02a0faf9-c5c7-4b47-b4e0-29c4d2906f90</Id>
                    <Name>ds3t2</Name>
                    <PermitGoingOutOfSync>false</PermitGoingOutOfSync>
                    <Quiesced>NO</Quiesced>
                    <ReplicatedUserDefaultDataPolicy />
                    <State>ONLINE</State>
                </Ds3Target>
                <Ds3Target>
                    <AccessControlReplication>NONE</AccessControlReplication>
                    <AdminAuthId>aid</AdminAuthId>
                    <AdminSecretKey>ask</AdminSecretKey>
                    <DataPathEndPoint>dp</DataPathEndPoint>
                    <DataPathHttps>true</DataPathHttps>
                    <DataPathPort />
                    <DataPathProxy />
                    <DataPathVerifyCertificate>true</DataPathVerifyCertificate>
                    <DefaultReadPreference>LAST_RESORT</DefaultReadPreference>
                    <Id>1da26e0c-40e2-4ee7-b68b-b60d07f70b51</Id>
                    <Name>ds3t3</Name>
                    <PermitGoingOutOfSync>false</PermitGoingOutOfSync>
                    <Quiesced>NO</Quiesced>
                    <ReplicatedUserDefaultDataPolicy />
                    <State>ONLINE</State>
                </Ds3Target>
            </Ds3Targets>

     Types that are not wrapped in a common element are parsed differently and do no require a separate function:
         <Delete>
             <Object><Key>object/</Key></Object>
             <Object><Key>object/1</Key></Object>
             <Object><Key>object/2</Key></Object>
         </Delete>

### CODE REVIEW

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java [hunk 1]
```diff
@@ -159,24 +163,56 @@ public void generateStaticFiles() throws IOException {
...
      */
     public static ImmutableSet<String> getArrayMemberTypes(final Ds3ApiSpec spec, final ImmutableSet<String> enumTypes) {
-            return spec.getTypes().values().stream()
-                    .flatMap(entry -> entry.getElements().stream())
-                    .filter(element -> element.getType().equalsIgnoreCase("array"))
-                    .filter(element -> !element.getComponentType().contains("UUID"))
-                    .filter(element -> !enumTypes.contains(EnumHelper.getDs3Type(element.getComponentType())))
-                    .map(element -> StructHelper.getResponseTypeName(element.getComponentType()))
-                    .collect(GuavaCollectors.immutableSet());
+        return spec.getTypes().values().stream()
```
Comment (RachelTucker) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java]: Some comments describing this stream/filtering would be helpful given its size.

File: ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java [hunk 2]
```diff
@@ -81,9 +81,9 @@ public boolean equals(final Object obj) {
             return false;
         }
         final Ds3Element element = (Ds3Element) obj;
-        return this.getName().equals(element.getName())
-                && this.getType().equals(element.getType())
-                && this.getComponentType().equals(element.getComponentType());
+        return Objects.equals(this.getName(), element.getName())
```
Comment (rpmoore) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: This is checking for reference equality, not string equality.

File: ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java [hunk 3]
```diff
@@ -81,12 +81,14 @@ public boolean equals(final Object obj) {
...
         }
         final Ds3Element element = (Ds3Element) obj;
-        return Objects.equals(this.getName(), element.getName())
-                && Objects.equals(this.getType(), element.getType())
-                && Objects.equals(this.getComponentType(), element.getComponentType());
+
+        return this.getName().equals(element.getName())
+                && this.getType() != null
+                && this.getType().equals(element.getType())
+                && this.getComponentType() != null
```
Comment (RachelTucker) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: what if both component types are null?

Comment (RachelTucker) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: types and component types are strings in Ds3Element

Comment (rpmoore) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: The String.equals method checks for null, so you only need to the check for null that you are calling `.equals` on.

Comment (DenverM80) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: All Ds3Elements have a name derived from either the Type or ComponentType (Rachel correct me if I'm wrong)

Comment (RachelTucker) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: Cannot type... Ds3Element is from the contract, so its as parsed from the contract

Comment (RachelTucker) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: Name, type, and component type are parsed directly from the contract. Of these, only component type is expected to be null sometimes.  Name is not derived.

Comment (DenverM80) [inline ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/apispec/Ds3Element.java]: gotcha.  will update.

**Pull request merged**